corehq/apps/reports/tasks.py

```diff
@@ -381,16 +401,19 @@ def find_question_id(form, value):
 
     for k, v in form.items():
         if isinstance(v, dict):
-            ret = find_question_id(v, value)
+            ret = _find_path_to_question_id(v, attachment_name, use_basename=use_basename)
             if ret:
                 return [k] + ret
         elif isinstance(v, list):
             for repeat in v:
-                ret = find_question_id(repeat, value)
+                ret = _find_path_to_question_id(repeat, attachment_name, use_basename=use_basename)
                 if ret:
                     return [k] + ret
         else:
-            if v == value:
+            if use_basename and os.path.isabs(v):
```

Comment:
I realize this was written in response to @millerdev 's feedback, but I'd like to push back a bit on readability/intuitiveness. Is there some easy way to describe what we're doing now? `use_basename` was appropriate because we relied on `os.path.basename` -- but `os.path.basename` treats absolute and relative paths identically. Because I can't think of any concise way to explain this edge case, and because I don't see the harm in treating relative paths the same way, I'd lean towards not checking whether the path is absolute.

Reply:
I think we only want to apply this transform in the very limited case where we get an absolute path since that is the known bug that motivated the change. Relative paths are not a concern for this fix (or even known to appear here), and seem more likely to indicate a bug somewhere in the system which should not pass silently. It seems better to do path adjustments as narrowly as possible to maintain a strict implementation with less room for messy submissions that automatically get "fixed" by this code.

My reasoning does violate the [robustness principle](https://en.wikipedia.org/wiki/Robustness_principle), but I think it makes the system less magical and is more in line with the [Zen](https://www.python.org/dev/peps/pep-0020/): Errors should never pass silently.

My understanding is we are working around the mobile bug this way because it is very hard to ensure that the fix is deployed to every device in the wild.

Reply:
I would argue, based on the prototype alone, that this makes the system more magical, not less. `use_basename=True` implies it always uses the familiar basename behavior, when it does not. When I can't think of some reasonable phrase to describe the current behavior, I take that as a sign that I should exercise caution around it.  That said, if you two disagree, I'm fine with being overruled.

Reply:
I agree with @millerdev that minimizing the impact of this change to only absolute paths is a good idea, but see your point @mjriley that this parameter name is not reflective of what actually happens. I can't think of a better parameter name without being too verbose, so I added some documentation around that parameter.

Can't answer if we typically have strict requirements around behaviors, though tests should do a reasonable job of this.